### PR TITLE
Add rds module

### PR DESCRIPTION
**Pull request by jr-nicola**
When I went through the app with Erik, he asked to add another tab so that users can download an RDS file. With the small sample we currently have, there aren't any rds files to download so I tested it with a demo .rds file I created and everything worked okay. I've also updated the {shinytest2} file to reflect this (structure of rds module is the similar to plots/tables tabs)

During testing, I also noticed the namespacing was missing from the downloadbutton so I've added this.

Depends on #10 being merged first

### CODE REVIEW

File: R/module_rds.R [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                  shiny::br(),
+                  shiny::fluidRow(
+                    shiny::column(12,
+                                  align = "center",
+                                  shiny::uiOutput(ns("download_rds_button")) #nolint
+                    )
+                  )
+  )
+}
+
```
Comment (russHyde) [inline R/module_rds.R:31]: The UI components are extremely similar between this module and the plots/tables modules. Could you add a function to define the wellPanels:

```r
display_panel = function(...) {
  shiny::wellPanel(
      shiny::fluidRow(
        shiny::column(
           12,
           align = "center",
           ...,
           style = "height:400px;"
        )
      ),
      style = "background: white"
   )
}
```
Then replace the different uses of this across the three modules.

An alterrnative would be to introduce a class on the 'column', and modify some CSS to set the align/style.

Comment (russHyde) [reply R/module_rds.R:31]: (Feel free to change the function name)

File: R/module_rds.R [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        dplyr::filter(.data$filetype %in% c("rds", "RDS")) %>%
+        dplyr::pull(.data$value)
+      rds_names = stringr::str_to_title(
+        stringr::str_replace_all(
+          gsub("\\..*", "", all_rds), "_", " "))
+      names(all_rds) = rds_names
+      shiny::selectInput(ns("rds_type"),
+                         label = "Select RDS file:",
+                         choices = all_rds)
+    })
```
Comment (russHyde) [inline R/module_rds.R:57]: There's some very similar code in the plots and tables modules.
Could you extract the inner code into a function and use that in module_[rds/plots/tables].

```r
filter_by_filetype = function(filenames, filetypes) {
  matching_files = filenames %>%
        dplyr::filter(.data$filetype %in% .env[["filetypes"]]) %>%
        dplyr::pull(.data$value)

 file_names = stringr::str_to_title(
        stringr::str_replace_all(
          gsub("\\..*", "", matching_files), "_", " "))

  names(matching_files) = file_names
  matching_files
}
```

**Review by russHyde — CHANGES_REQUESTED** [review]
Hi. It's mostly good, but as a result of having introduced a third module that is similar to `module_plots` / `module_tables` we're getting a lot of duplication popping up. For now, it would be easier to introduce functions to pay-down that duplication, rather than writing a more general purpose server/UI pair.

File: R/module_rds.R (hunk 2 above)
Comment (jr-nicola) [reply R/module_rds.R]: Thanks - I've added this function, and updated the code inside the modules.

File: R/module_rds.R (hunk 1 above)
Comment (jr-nicola) [reply R/module_rds.R]: I've added in this function and switched the code in the modules.

**Review by russHyde — APPROVED** [review]
This is great. Thanks

**Pull request merged**